src/explorer/nodeBase.ts

```diff
@@ -20,11 +19,17 @@ export class NodeBase {
         };
     }
 
-    async getChildren(azureAccount: AzureAccountWrapper): Promise<NodeBase[]> {
-        return [];
+    getTreeDataProvider<T extends TreeDataProvider<NodeBase>>(): T {

```

Comment:
I think we can get rid of 'T' here. It will always be the same type in this codebase